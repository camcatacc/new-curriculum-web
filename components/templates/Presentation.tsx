import React from "react";
import { SectionTitle } from "../molecules/SectionTitle";
import { About } from "../organisms/About";

export const Presentation = (props: { className?: string }) => {
    return (
        <div className={`flex flex-col justify-center ${props.className}`}>
            <SectionTitle>QUI SÃ“C</SectionTitle>
            <About name="David" surname="Camacho Cateura" />
        </div>
    );
};
