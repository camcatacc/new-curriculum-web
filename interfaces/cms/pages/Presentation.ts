// Constants
export const CONTENT_TYPE_PRESENTATION_PAGE = "aboutPage";

// Definitions
import type { Document } from "@contentful/rich-text-types";

export interface CmsPresentationPage {
	title: string;
	aboutText: Document;
	name: string;
	surname: string;
}
