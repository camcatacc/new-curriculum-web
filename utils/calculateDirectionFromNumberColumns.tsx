// TODO: Simplify this function
function calculateDirectionFromNumberColumns(i: number, total: number, numPerRow: number): "fromLeft" | "fromRight" | "fromBottom" {
	// If everything does not fit in a row
	if (numPerRow < total) {
		// We calculate the number of items in the last row
		let nItemsLastRow = total % numPerRow;
		// We calculate the position from which the items will be in the last row
		let threshold = total - nItemsLastRow;
		// And we calculate the item position in that row
		let posInNewRow = i - numPerRow;

		// If the threshold is smaller than the item position and there are remaining items to be placed enter this conditional
		// If not, then the default method can be applied
		if (threshold < i && nItemsLastRow !== 0) {
			// If there is only one item in the last row, it must come from the bottom
			if (nItemsLastRow === 1) return "fromBottom";
			// We use the default method BUT with the position of the item IN THE LAST ROW
			switch (posInNewRow % nItemsLastRow) {
				case 0:
					return "fromRight";
				case 1:
					return "fromLeft";
				default:
					return "fromBottom";
			}
		}
	}

	switch (
		i % numPerRow //
	) {
		case 0:
			return "fromRight";
		case 1:
			return "fromLeft";
		default:
			return "fromBottom";
	}
}

export default calculateDirectionFromNumberColumns;
