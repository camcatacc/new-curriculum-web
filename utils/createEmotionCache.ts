// Modules
import createCache from "@emotion/cache";

// Functions
export default function createEmotionCache() {
	return createCache({ key: "css" });
}
