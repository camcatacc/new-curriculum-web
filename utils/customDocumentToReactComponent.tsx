// Modules
import React from "react";
import { BLOCKS, MARKS } from "@contentful/rich-text-types";
import { documentToReactComponents } from "@contentful/rich-text-react-renderer";

// Styles
import { secondary } from "styles/customColors";

// Elements
import { Paragraph } from "@hardru/curriculum-web-components";

// Definitions
import type { Document } from "@contentful/rich-text-types";
import type { Options } from "@contentful/rich-text-react-renderer";
import type { PropTypes, Theme, TypographyVariant } from "@material-ui/core";

export interface CustomDocumentToReactComponentOptions {
	theme: Theme;
	boldColor?: boolean;
	marginBottom?: 1 | 2 | 3 | 4 | 6 | 8 | 10;
	align?: PropTypes.Alignment;
	variant?: TypographyVariant;
}

// Function
function customDocumentToReactComponent(document: Document, { boldColor, align, variant, theme }: CustomDocumentToReactComponentOptions) {
	let options: Options = {
		renderNode: {
			[BLOCKS.PARAGRAPH]: (_: any, children: any) => (
				<div>
					<Paragraph variant={variant} align={align}>
						{children}
					</Paragraph>
				</div>
			)
		}
	};

	if (boldColor) {
		options = { ...options, renderMark: {} };
		options.renderMark![MARKS.BOLD] = (text) => <b style={{ color: theme.palette.secondary[theme.palette.type] }}>{text}</b>;
	}

	return documentToReactComponents(document, options);
}

export default customDocumentToReactComponent;
