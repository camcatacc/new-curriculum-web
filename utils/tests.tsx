// Modules
import React, { ReactElement } from "react";
import { ThemeProvider } from "@material-ui/styles";

// Testing modules
import { render, RenderOptions, RenderResult } from "@testing-library/react";

// Redux Modules
import { configureStore } from "@reduxjs/toolkit";

// Redux Components
import rootReducer from "redux/rootReducer";
import { Provider } from "react-redux";
import { defaultTestRootState } from "redux/store.test";

// Auxiliary functions
import { getMaterialTheme } from "styles/materialUi";

// Definitions
import type { RootState } from "redux/types";

// Functions
export const customRender = (ui: ReactElement, state: Partial<RootState> = defaultTestRootState, renderOptions?: RenderOptions & { darkMode?: boolean }): RenderResult => {
	const Wrapper: React.FC = ({ children }) => {
		const store = configureStore({
			reducer: rootReducer,
			preloadedState: state
		});

		return (
			<ThemeProvider theme={getMaterialTheme(renderOptions?.darkMode ? "dark" : "light")}>
				<Provider store={store}>{children}</Provider>
			</ThemeProvider>
		);
	};

	return render(ui, { wrapper: Wrapper, ...renderOptions });
};
