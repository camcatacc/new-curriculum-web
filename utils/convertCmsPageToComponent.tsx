// Modules
import React from "react";

// Elements
import Abilities from "components/templates/Abilities/Abilities";
import Presentation from "components/templates/Presentation/Presentation";
import FindMe from "components/templates/FindMe/FindMe";

// Definitions
import type { ContentfulEntry } from "interfaces/cms/contentful";
import type { CmsPage } from "interfaces/cms/pages/Page";
import type { CmsPresentationPage } from "interfaces/cms/pages/Presentation";
import type { CmsAbilitiesPage } from "interfaces/cms/pages/Abilities";
import type { CmsFindMePage } from "interfaces/cms/pages/FindMe";

// Functions
const convertCmsPageToComponent = (page: ContentfulEntry<CmsPage>) => {
	switch (page.fields.type) {
		case "PresentationPage":
			return presentationCmsToComponent(page.fields.content.fields);
		case "AbilitiesPage":
			return abilitiesCmsToComponent(page.fields.content.fields);
		case "FindMePage":
			return findMeCmsToComponent(page.fields.content.fields);
	}
};

const presentationCmsToComponent = (page: CmsPresentationPage) => <Presentation {...page} />;
const abilitiesCmsToComponent = (page: CmsAbilitiesPage) => {
	const lists = page.abilitiesLists.map((abList) => {
		return {
			...abList.fields
		};
	});

	return <Abilities title={page.title} abilitiesLists={lists} />;
};

const findMeCmsToComponent = (page: CmsFindMePage) => <FindMe {...page} />;

export default convertCmsPageToComponent;
