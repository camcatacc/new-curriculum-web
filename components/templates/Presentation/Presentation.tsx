// Modules
import React from "react";

// Elements
import SectionTitle from "components/molecules/SectionTitle/SectionTitle";
import About from "components/organisms/About/About";
import { CmsPresentationPage } from "interfaces/cms/pages/Presentation";

// Definitions
export interface PresentationProps extends CmsPresentationPage {
	className?: string;
}

// Element
const Presentation = ({ className, title, aboutText, name, surname }: PresentationProps) => {
	return (
		<div className={`flex flex-col justify-center ${className}`}>
			<SectionTitle animated>{title}</SectionTitle>
			<About name={name} surname={surname} paragraphsDoc={aboutText} animated />
		</div>
	);
};

export default Presentation;
