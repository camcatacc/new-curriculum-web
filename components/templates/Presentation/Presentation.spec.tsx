// Modules
import React from "react";
import { BLOCKS } from "@contentful/rich-text-types";

// Test Modules
import { customRender } from "utils/tests";

// Accessibility Tests
import { axe, toHaveNoViolations } from "jest-axe";

// Components
import Presentation, { PresentationProps } from "components/templates/Presentation/Presentation";

// Accessibility extension
expect.extend(toHaveNoViolations);

// Default Components
const defaultProps: PresentationProps = {
	name: "David",
	surname: "Camacho",
	title: "About me",
	aboutText: {
		data: {},
		content: [
			{
				data: {},
				content: [
					{
						data: {},
						marks: [],
						value: "I am a ",
						nodeType: "text"
					},
					{
						data: {},
						marks: [
							{
								type: "bold"
							}
						],
						value: "commited",
						nodeType: "text"
					},
					{
						data: {},
						marks: [],
						value: ", capable and resolutive person. I have skills, experience and formation as a ",
						nodeType: "text"
					},
					{
						data: {},
						marks: [
							{
								type: "bold"
							}
						],
						value: "developer",
						nodeType: "text"
					},
					{
						data: {},
						marks: [],
						value: ", which are very well complemented by my ",
						nodeType: "text"
					},
					{
						data: {},
						marks: [
							{
								type: "bold"
							}
						],
						value: "business background",
						nodeType: "text"
					},
					{
						data: {},
						marks: [],
						value: ".",
						nodeType: "text"
					}
				],
				nodeType: BLOCKS.PARAGRAPH
			},
			{
				data: {},
				content: [
					{
						data: {},
						marks: [],
						value: "I led an educational robotics and programming firm for five years. I devoted the last one and a half to ideate, design and develop an ",
						nodeType: "text"
					},
					{
						data: {},
						marks: [
							{
								type: "bold"
							}
						],
						value: "online solution in the form of a hybrid app",
						nodeType: "text"
					},
					{
						data: {},
						marks: [],
						value: " and a web platform. ",
						nodeType: "text"
					}
				],
				nodeType: BLOCKS.PARAGRAPH
			},
			{
				data: {},
				content: [
					{
						data: {},
						marks: [],
						value: "I am currently employed by Accenture as a ",
						nodeType: "text"
					},
					{
						data: {},
						marks: [
							{
								type: "bold"
							}
						],
						value: "full-stack developer",
						nodeType: "text"
					},
					{
						data: {},
						marks: [],
						value: ".",
						nodeType: "text"
					}
				],
				nodeType: BLOCKS.PARAGRAPH
			}
		],
		nodeType: BLOCKS.DOCUMENT
	}
};

// Custom rendering
function renderComponent(newProps?: object) {
	const props = { ...defaultProps, ...newProps };
	return customRender(<Presentation {...props} />);
}

describe("Presentation", () => {
	it("Snapshot and accessibility tests", async () => {
		const { container } = renderComponent();
		expect(container).toMatchSnapshot();
		expect(await axe(container)).toHaveNoViolations();
	});
});
