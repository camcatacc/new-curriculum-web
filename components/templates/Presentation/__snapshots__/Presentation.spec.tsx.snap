// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Presentation Snapshot and accessibility tests 1`] = `
<div>
  <div
    class="flex flex-col justify-center undefined"
  >
    <div
      class="flex flex-col mt-8 mb-8"
    >
      <div>
        <h2
          class="MuiTypography-root uppercase MuiTypography-h2 MuiTypography-alignCenter"
        >
          About me
        </h2>
      </div>
      <div
        class="self-center border-b-2 border-black border-solid w-20 mt-4 mb-4"
      />
    </div>
    <div
      class="sc-kDTinF iQGOnz flex flex-col md:flex-row items-center"
    >
      <div
        class="sc-iqseJM iVBSQY w-3/5 md:w-auto"
      >
        <div
          class="sc-jrQzAO CEymC w-full relative mb-4"
        >
          <img
            alt="DavidCamacho"
            class="absolute top-0 left-0 bottom-0 right-0 rounded-full"
            src="/avatar.jpg"
          />
        </div>
        <h3
          class="MuiTypography-root sc-dkPtRN uHNAI MuiTypography-h3 MuiTypography-alignCenter"
        >
          David
           
          Camacho
        </h3>
      </div>
      <div
        class="sc-crHmcD fHBznV mt-4 md:mt-0 flex gap-8 flex-col justify-center"
      >
        <div>
          <p
            class="MuiTypography-root sc-pVTFL ctltIR MuiTypography-body1 MuiTypography-alignJustify"
          >
            I am a 
            <b
              style="color: rgb(0, 2, 67);"
            >
              commited
            </b>
            , capable and resolutive person. I have skills, experience and formation as a 
            <b
              style="color: rgb(0, 2, 67);"
            >
              developer
            </b>
            , which are very well complemented by my 
            <b
              style="color: rgb(0, 2, 67);"
            >
              business background
            </b>
            .
          </p>
        </div>
        <div>
          <p
            class="MuiTypography-root sc-pVTFL ctltIR MuiTypography-body1 MuiTypography-alignJustify"
          >
            I led an educational robotics and programming firm for five years. I devoted the last one and a half to ideate, design and develop an 
            <b
              style="color: rgb(0, 2, 67);"
            >
              online solution in the form of a hybrid app
            </b>
             and a web platform. 
          </p>
        </div>
        <div>
          <p
            class="MuiTypography-root sc-pVTFL ctltIR MuiTypography-body1 MuiTypography-alignJustify"
          >
            I am currently employed by Accenture as a 
            <b
              style="color: rgb(0, 2, 67);"
            >
              full-stack developer
            </b>
            .
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
